### PR TITLE
fix: check for permissions before updating webhooks

### PR DESCRIPTION
**Pull request by bernot-dev**
In preparation for removing permissions from GKE environments, add an explicit check to ensure updating CA Bundle on webhooks is allowed before blindly attempting to do so forever.

### CODE REVIEW

File: pkg/operator/operator.go [hunk 1]
```diff
@@ -606,3 +588,63 @@ func (o *Operator) setMutatingWebhookCABundle(ctx context.Context, caBundle []by
...
+		authv1.ResourceAttributes{
+			Group:     arv1.GroupName,
+			Resource:  "ValidatingWebhookConfiguration",
+			Namespace: o.opts.OperatorNamespace,
+			Verb:      "update",
+		},
+	)
+}
+
+func (o *Operator) canI(ctx context.Context, resources ...authv1.ResourceAttributes) (bool, error) {
```
Comment (macxamin) [inline pkg/operator/operator.go:609]: nit: Can you make the name of this function more explicit? maybe merge with `canUpdateWebhooks`.

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -606,3 +588,63 @@ func (o *Operator) setMutatingWebhookCABundle(ctx context.Context, caBundle []by
...
+	for _, resource := range resources {
+		req := authv1.SelfSubjectAccessReview{
+			Spec: authv1.SelfSubjectAccessReviewSpec{
+				ResourceAttributes: &resource,
+			},
+		}
+		if err := o.client.Create(ctx, &req); err != nil {
```
Comment (TheSpiritXIII) [inline pkg/operator/operator.go:616]: **thought (non-blocking):** This is cool! I didn't realize that you could check permissions this way. Do you know how this compares to something like a dry-run client?

File: pkg/operator/operator.go [hunk 3]
```diff
@@ -606,3 +588,63 @@ func (o *Operator) setMutatingWebhookCABundle(ctx context.Context, caBundle []by
...
+			return false, fmt.Errorf("check permissions to %q %q: %w", resource.Verb, resource.Name, err)
+		}
+		if !req.Status.Allowed {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (o *Operator) continuouslySetCABundle(ctx context.Context, caBundle []byte) {
```
Comment (TheSpiritXIII) [inline pkg/operator/operator.go:626]: **note:** link to non-polling solution for this: https://github.com/GoogleCloudPlatform/prometheus-engine/pull/350

**Review by TheSpiritXIII — APPROVED** [review]
Thanks!

File: pkg/operator/operator.go (hunk 1 above)
Comment (bernot-dev) [reply pkg/operator/operator.go:609]: canUpdateWebhooks is on `:592`

`canI` is a generic implementation based on the same idea as [kubectl auth can-i](https://kubernetes.io/docs/reference/access-authn-authz/authorization/#checking-api-access).

File: pkg/operator/operator.go (hunk 2 above)
Comment (bernot-dev) [reply pkg/operator/operator.go:616]: `can-i` just checks for permissions without checking for any other kind of API conformity, like a dry-run client might.

**Pull request merged**